module1/homework/lengthSort/sort.cpp

```diff
@@ -0,0 +1,18 @@
+#include "sort.hpp"
+
+std::deque<std::string> lengthSort(const std::forward_list<std::string>& list) {
+    std::deque<std::string> words_deq;
+
+    auto deq_it = words_deq.begin();
+    for(auto& word : list) {
+        deq_it = words_deq.begin();
+        while(deq_it != words_deq.end()) {
+            if(word.size() < deq_it->size() || (word.size() == deq_it->size() && word < *deq_it)) {
+                break;
+            }
```

Comment:
You don't modify word in list so you can use const auto& in for loop :)
```suggestion
    for(const auto& word : list) {
        deq_it = words_deq.begin();
        while(deq_it != words_deq.end()) {
            if(word.size() < deq_it->size() || (word.size() == deq_it->size() && word < *deq_it)) {
                break;
            }
```